raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2083,6 +2098,41 @@ private boolean handleApiVersionsResponse(
         );
     }
 
+    private CompletableFuture<RemoveRaftVoterResponseData> handleRemoveVoterRequest(
+        RaftRequest.Inbound requestMetadata,
+        long currentTimeMs
```

Comment:
I guess this applies to addVoter as well, but we really should be using the monotonic clock here, not the wall clock, in case someone adjusts the wall clock.

Maybe should be done in a follow-up.

Reply:
Good point. Let me file a jira to revisit KRaft's source of time. In some cases it needs to be a wall-clock. Like when it it is persisted to disk as the leader's append time. But for the timers, it doesn't need to be based on a wall-clock.

Reply:
https://issues.apache.org/jira/browse/KAFKA-17196